diff --git a/web/.htaccess b/web/.htaccess
index 7fedd47..fdf1dd1 100644
--- a/web/.htaccess
+++ b/web/.htaccess
@@ -65,6 +65,10 @@ AddEncoding gzip svgz
   RewriteCond %{HTTP:X-Forwarded-Proto} !https
   RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
 
+  # Redirect old web name to new web name.
+  RewriteCond %{HTTP_HOST} ^ucommphotosrv\.unl\.edu$ [NC]
+  RewriteRule ^(.*)$ https://ucommphotos.unl.edu/ [R=301,L]
+
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
